src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx

```diff
@@ -197,3 +181,115 @@ export const cardSidePadding = css`
 const TabContentWrapper = styled.div`
   ${cardSidePadding}
 `;
+
+type CheckboxState = "checked" | "unchecked" | "indeterminate";
+
+const getTaskCheckboxState = (
+  taskName: string,
+  selectedBuildVariants: string[],
+  selectedBuildVariantTasks: VariantTasksState,
+  variants: ProjectBuildVariant[]
+): CheckboxState => {
+  const checkedCb = (allChecked: boolean, buildVariantName: string): boolean =>
+    allChecked &&
+    !!get(selectedBuildVariantTasks, [buildVariantName, taskName], false);
```

Comment:
i propose not using `get` going forward because it does not catch type errors at compile time. from the PR summary saying the bug was "caused by passing an improperly formatted 2nd parameter to lodash.get", it sounds like type safety with get could have been the issue?

`!!get(selectedBuildVariantTasks, [buildVariantName, taskName], false);` would become 
```
selectedBuildVariantTasks.includes(buildVariantName) && selectedBuildVariantTasks.includes(taskName)
```
or something like that

Reply:
this is so true. `get` also doesn't always infer the correct return type from the first parameter when accessing nested values

Reply:
for accessing props in gql data, we can do `const somethingFromData = data?.somethingFromData ?? []` which will do a similar thing to `get` while utilizing type safety